### ISSUE TITLE
Add is valid geometry constraint

### ISSUE THREAD

**Issue by m-kuhn** [QGEP/QGEP#382]
Invalid geometries with self intersection, duplicated nodes (and possibly others) are not accepted to be processed by ili2pg export.
To avoid such issues from the beginning, a constraint including `ST_IsValid` should be added on geometries.

**Comment by ponceta** [comment]
@m-kuhn having a validity constraint on object geometry is a good practice? 
In terms of performance, how is the impact on the db operations?

**Comment by m-kuhn** [comment]
With new insights I am a bit hesitant on this approach.

* The database will reject the geometry in editing sessions without giving the user any chance to fix it (except for redigitizing all).
* There are other things that can go wrong, e.g. on ili2pg export the precision attribute is taken into account, which leads to "snapping" of coordinates which in turn can also introduce duplicate coordinates and problems.
* I don't think performance will be affected noticeably (it's only on feature insert and the network roundtrip and I/O will be more expensive than this check)

I think we'll rather want to have this as an additional check and/or automated process which runs on export.

**Comment by haubourg** [comment]
Indeed, agreed with Matthias. I generally propose soft warnings queries displayed in QGIS. This shows immediatly any issue when digitzing without rejecting the commit. 
A folder with all possible soft warning layers is a good pratice, only editors will use it, and you can enrich your project with your own business logic checks using dmanager views without necessarily storing the views inside QGEP (and open the can of worm of local customizations )

**Comment by ponceta** [comment]
To access geometric validity, I would recommend to create both additionnal approaches:

-  Create views to detect topological errors which can be corrected by the users
-  Check geom validity in export views

**Comment by m-kuhn** [comment]
We have been discussing this yesterday with other Ili2db users. The most annoying and most frequent issue is the combination of precision and duplicate points. The important thing to note here is, that without the additional interlis precision definitions, the geometries are completely correct, and therefore impossible to detect.

E.g.
User digitizes in high precision (tracing etc.) these points are then snapped to a lower precision (e.g. snapped to 1 cm by ili2pg because of model definitions) and therefore duplicate points are the result.

What we plan to do there is to "fix" this as early as possible. The idea is to have new layer options for automatic fixing of common problems. In this case on any geometry addition/update a "snap to grid" and "remove duplicate vertices" would be run, so the user sees the final result he will have in the interlis file immediately.

The main counter argument to this is: do you need to be able to work in a higher precision than in the interlis definition?
In this case, let's look for alternative solutions!
